plugins/defaultplugins/l3plugin/route_config.go

```diff
@@ -44,14 +45,18 @@ type RouteConfigurator struct {
 	RouteIndexes  idxvpp.NameToIdxRW
 	RouteIndexSeq uint32
 	SwIfIndexes   ifaceidx.SwIfIndex
-	vppChan       *govppapi.Channel
-	Stopwatch     *measure.Stopwatch // timer used to measure and store time
+	RouteCachedIndex idxvpp.NameToIdxRW
```

Comment:
handle cache the route is deleted (remove the route from the cache if it is there)
also you should use custom type (for example RouteIndex instead of NameToIdxRW - see comments below)